docs/src/pages/components/progress/CircularIntegration.tsx

```diff
@@ -46,15 +46,15 @@ export default function CircularIntegration() {
   const classes = useStyles();
   const [loading, setLoading] = React.useState(false);
   const [success, setSuccess] = React.useState(false);
-  const timer = React.useRef<number>();
+  const timer = React.useRef<ReturnType<typeof setTimeout>>();
```

Comment:
Downside of a structural type system: TS has no idea that we only ever assign to it in a browser where this would be `number` and not a nodejs timeout. The upside is that this code is easier to integrate into an existing nextjs codebase.

Reply:
What do you think about this approach? https://github.com/mui-org/material-ui/pull/22697/commits/1b87739cbcde376e20de37f1bc456eb1ca6ba4c9

Reply:
I have seen it suggested in https://github.com/Microsoft/TypeScript/issues/30128#issuecomment-467968688.

Reply:
`window` makes sense. https://github.com/microsoft/TypeScript/issues/30128#issuecomment-467968688 not so much since this is backwards (`global` is not available in a browser).

Reply:
What about the perf impact of property access on the `window` object though? :laughing: